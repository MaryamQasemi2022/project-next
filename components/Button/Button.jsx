import React from "react";
import { css,} from '@emotion/css';
import {  useTheme } from '@emotion/react'
import Img from "../Img/Img";

const Button=({
  children,
  type='button',
  widthAll,
  loading=false,
  disabled=false,
  onClick=()=>{}
})=>{
  const theme=useTheme()
  return(
   
       <button onClick={onClick} disabled={disabled} type={type}  className={css`

       width:100%;
       color:${type=='primary'?'white':'white'};
        background-color:${theme.colors.blue};
        font-size:${theme.typography.paragraph.fontSize}rem;
        padding:${theme.spaces[4]};
        border-radius:${theme.borderRadius[1]};
        border:none;
        display:flex;
        justify-content:center;
        cursor:pointer;
        position:relative;
        transition: box-shadow 0.5s;
        box-shadow:${theme.boxShadow.light};
        &:hover{
          box-shadow:none;
          background-color:${theme.colors.secondary};
        
        }
        `} >
      {loading ? <Img width='30px' height='25px' src='loading/loading.gif'/>: children}
    </button>
  
  )
}

export default Button;
