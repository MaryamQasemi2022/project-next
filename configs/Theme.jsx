const Theme={
  colors:{
    primary:'#f7f7f7',//gray
    secondary:'#f77c38',//orange
    blue:"#68e1fd",//blue

  },
  typography:{
      h1:{
        fontSize:1.802,
        fontWeight:500,
        lineHeight:1.5
      },
      h2:{
        fontSize:1.602,
        fontWeight:500,
        lineHeight:1.5
      },
      h3:{
        fontSize:1.424,
        fontWeight:500,
        lineHeight:1.5
      },
      h4:{
        fontSize:1.266,
        fontWeight:500,
        lineHeight:1.5
      },
      h5:{
        fontSize:1.125,
        fontWeight:500,
        lineHeight:1.5
      },
      paragraph:{
        fontSize:1.256,
        fontWeight:300,
        lineHeight:1.15
      },
      small:{
        fontSize:0.889,
        fontWeight:100,
        lineHeight:1.15
      }
  },
    spaces:['0px','2px','4px','8px','16px','32px','64px'],
    borderRadius:['0px','2px'],
    borderColor:{
      light:'#e3e4e6'
    },
    boxShadow:{
      light:'0 4px 16px rgba(0,1,31,0.2)'
    }
  
}

export default Theme;
