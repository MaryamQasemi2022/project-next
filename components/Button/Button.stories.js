

import Button from './Button';

export default {

  title: 'Button',
  component: Button,
};

export const Primary = () => <Button type="primary"> ورود</Button>;
export const Secondary = () => <Button type="secondary"> ارسال</Button>;
export const WidthAll = () => <Button widthAll> click me</Button>;


