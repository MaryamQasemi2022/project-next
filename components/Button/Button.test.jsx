/**
 * @jest-environment jsdom
 */
import React from "react";
import Button from "./Button";
import mountWidthTheme from "../../utils/mountWithTheme"
import { render,fireEvent} from '@testing-library/react';
// import renderer from 'react-test-renderer';
import { ThemeProvider } from '@emotion/react'
import Theme from '../../configs/Theme';

describe('button component', () => {
  test('render correctly', () => {
    const component =mountWidthTheme(<Button type="primary">sample button</Button>)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test(' with loading ', () => {
    const component =mountWidthTheme(<Button loading >loading button</Button>)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test(' with disable', () => {
    const component =mountWidthTheme(<Button disabled >disable button</Button>)
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  // test(' with click', () => {
  //   let variable=0;
  //   const {queryByText, getByText} = render(
  //     <ThemeProvider theme={Theme}>
  //       <Button  onClick={()=>{
  //         variable +=1
  //       }}>
  //         click here
  //         </Button>
  //     </ThemeProvider>
  //   );
  //   fireEvent.click(getByText(/click/))

  //   expect(variable).toBe(0);
  // })
});

it('check click button',()=>{
    let variable="not click";
    const {queryByText, getByText} = render(
      <ThemeProvider theme={Theme}>
        <Button  onClick={()=>{
          variable="clicked"
        }}>
          click
          </Button>
      </ThemeProvider>
    );
    expect(queryByText(/click/)).toBeTruthy();
    fireEvent.click(queryByText(/click/))

    expect(queryByText(/click/)).toBeTruthy();
  
})

