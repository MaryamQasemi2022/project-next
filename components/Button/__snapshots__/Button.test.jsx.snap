// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component  with disable 1`] = `
<button
  className="css-1avd6i"
  disabled={true}
  onClick={[Function]}
  type="button"
>
  disable button
</button>
`;

exports[`button component  with loading  1`] = `
<button
  className="css-1avd6i"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <div
    className="css-1hailib"
  >
    <img
      alt="picture"
      className="css-1ki26gy"
      src="loading/loading.gif"
    />
  </div>
</button>
`;

exports[`button component render correctly 1`] = `
<button
  className="css-1avd6i"
  disabled={false}
  onClick={[Function]}
  type="primary"
>
  sample button
</button>
`;
